Generated JAR does raises java.lang.ClassNotFoundException: io.etcd.jetcd.ByteSequence

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Clone this repo https://github.com/marcosflobo/micronaut-etcd
2. Create the jar `./gradlew clean assemble`
3. Create a simple Java Gradle app and import the jar generated in step 2
4. Run in a docker container the `etcd` server (see below)
5. Fill in the main method (see Example Java Gradle Application) and run the app

### Expected Behaviour

The Java Gradle app should finish properly and do not break

### Actual Behaviour

The Java Gradle App raises an error about ClassNotFound
```
Exception in thread "main" java.lang.NoClassDefFoundError: io/etcd/jetcd/ByteSequence
	at io.micronaut.etcd.config.EtcdFactoryConfig.<init>(EtcdFactoryConfig.java:42)
	at io.micronaut.etcd.config.SingleEtcdFactoryConfig.<init>(SingleEtcdFactoryConfig.java:36)
	at Main.main(Main.java:9)
Caused by: java.lang.ClassNotFoundException: io.etcd.jetcd.ByteSequence
Caused by: java.lang.ClassNotFoundException: io.etcd.jetcd.ByteSequence

	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 3 more

Execution failed for task ':Main.main()'.
> Process 'command '/usr/lib/jvm/java-1.11.0-openjdk-amd64/bin/java'' finished with non-zero exit value 1
```

### Environment Information

- **Operating System**: Ubuntu 20.04
- **Micronaut Version:** 2.0.2
- **JDK Version:** 11

### Example environment
First, run the `etcd` server and then the Java app.
#### Run etcd docker container
```
docker run -it --rm -p 2379:2379 -p 4001:4001 --name myetcd quay.io/coreos/etcd:v3.4.9 /usr/local/bin/etcd -advertise-client-urls http://0.0.0.0:2379 -listen-client-urls http://0.0.0.0:2379
```
#### Example Java Gradle Application
##### build.gradle
```
plugins {
    id 'java'
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    compile files('/path/to/micronaut-etcd/etcd/build/libs/etcd-1.0.0.BUILD-SNAPSHOT.jar')
    testCompile group: 'junit', name: 'junit', version: '4.12'
}
```
##### Main.java
```
import io.micronaut.etcd.config.EtcdFactoryConfig;
import io.micronaut.etcd.config.SingleEtcdFactoryConfig;
import io.micronaut.etcd.kv.KVService;
import java.util.concurrent.ExecutionException;

public class Main {

  public static void main(String[] args) throws ExecutionException, InterruptedException {
    EtcdFactoryConfig config = new SingleEtcdFactoryConfig();
    config.setEndpoints("http://localhost:2379");
    KVService kvService = new KVService(config);
    byte[] ret = kvService.get("key");
  }
}
```

--- a/etcd/build.gradle
+++ b/etcd/build.gradle
@@ -2,9 +2,9 @@ dependencies {
     annotationProcessor "io.micronaut:micronaut-inject-java"
     annotationProcessor "io.micronaut.docs:micronaut-docs-asciidoc-config-props:$micronautDocsVersion"
 
-    api "io.micronaut:micronaut-inject"
-    api "io.micronaut:micronaut-inject-java"
-    compile "io.etcd:jetcd-core:$jetcdVersion"
+    implementation "io.micronaut:micronaut-inject"
+    implementation "io.micronaut:micronaut-inject-java"
+    implementation "io.etcd:jetcd-core:$jetcdVersion"
 
     compileOnly "io.micronaut.micrometer:micronaut-micrometer-core"
     compileOnly "io.micronaut:micronaut-management"

